test/Object/getOwnPropertyDescriptors.js

```diff
@@ -0,0 +1,89 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+
+if (this.WScript && this.WScript.LoadScriptFile) { // Check for running in ch
+    this.WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+}
+
+var tests = [
+    {
+        name: "Object has getOwnPropertyDescriptors method",
+            body: function() {
+                assert.isTrue(Object.hasOwnProperty("getOwnPropertyDescriptors"), 'Object.hasOwnProperty("getOwnPropertyDescriptors")');
+
+            }
+    },
+    {
+        name: "Correctly handles bad parameters.",
+        body: function() {
+            assert.throws(function() {
+                WScript.Echo(Object.getOwnProperties());
+            }, TypeError, "");
+
+            assert.throws(function() {
+                WScript.Echo(Object.getOwnProperties(null));
+            }, TypeError, "");
+        }
+    },
+    {
+        name: "The resulting get and set are identical with the original get and set.",
+        body: function() {
+            // This test is taken from https://github.com/tc39/proposal-object-getownpropertydescriptors/blob/master/test/built-ins/Object/getOwnPropertyDescriptors/has-accessors.js
+
+
+            var a = {
+                get a() {},
+                set a(value) {}
+            };
+            var b = Object.getOwnPropertyDescriptors(a);
+
+
+            assert.isTrue(b.a.get === Object.getOwnPropertyDescriptor(a, 'a').get);
+            assert.isTrue(b.a.set === Object.getOwnPropertyDescriptor(a, 'a').set);
+        }
+    },
+    {
+        name: "The list of property descriptors includes all own properties with correct descriptors",
+        body: function() {
+            var foo = {}
+
+            Object.defineProperties(foo, {
+                "fooAllTrue": {
+                    configurable: true,
+                    enumerable: true,
+                    value: "fooAllTrue",
+                    writable: true
+                },
+                "fooAllFalse": {
+                    configurable: false,
+                    enumerable: false,
+                    value: "fooAllFalse",
+                    writable: false
+                }

```

Comment:
Should also have some properties with symbol names and with integer names, since these property names are handled differently than string names.